src/python/pants/backend/go/util_rules/build_pkg.py

```diff
@@ -171,6 +179,45 @@ class BuiltGoPackage:
     import_paths_to_pkg_a_files: FrozenDict[str, str]
 
 
+@dataclass
+@frozen_after_init
+class EmbedConfig:
+    """Configuration passed to the compiler to configure file embedding. The compiler relies
+    entirely on the caller to map file patterns to actual filesystem paths.
+
+    patterns: Maps a pattern to a list of _relative_ file paths to use for that pattern. These relative file paths
```

Comment:
It'd be helpful to define relative, given that files are always relative in Pants just relative to the buildroot.

Reply:
Comment updated. They are used to define the virtual directory hierarchy exposed when the embedding variable types is `embed.FS` (which is a virtual filesystem).